grails-app/assets/javascripts/spApp/directive/selectSpecies.js

```diff
@@ -150,9 +151,48 @@
                             LayoutService.openModal('tool', {processName: 'ToolAddSpeciesService'})
                         };
 
+                        // Old values only has the change of 'changing' item in AngularJs<1.7
+                        // thus cannot simply rollback to old values
+                        // scope.$watchGroup(['spatiallyValid','spatiallySuspect','spatiallyUnknown'], function(newValues, oldValues) {
+                        //     if(newValues.every(function(x) { return !x; })){
+                        //         alert('Select at least one spatial related options!')
+                        //         scope.spatiallyValid = oldValues[0];
+                        //         scope.spatiallySuspect = oldValues[1];
+                        //         scope.spatiallyUnknown = oldValues[2];
+                        //     }
```

Comment:
delete commented out code

Reply:
I prefer leaving it there, otherwise someone may question why we use this clumsy way to solve issue?

Reply:
OK - comment is not clear to me, so I suggest also making the comment clearer. 
>Old values only has the change of 'changing' item in AngularJs<1.7
I don't know what `only has the change of 'changing' item` means - rewrite this to be really explicit.